drgn_tools/task.py

```diff
@@ -197,6 +197,91 @@ def get_command(task: Object) -> str:
     return escape_ascii_string(task.comm.string_())
 
 
+def get_ppid(task: Object) -> int:
+    """
+    :returns: Parent PID of the task
+    """
+    return task.parent.pid.value_()
+
+
+def get_rss(prog: Program, task: Object) -> int:
+    """
+    :returns RSS value
+    """
+    try:
+        filerss = task.mm.rss_stat.count[0].counter.value_()
+        if filerss < 0:
+            filerss = 0
+        anonrss = task.mm.rss_stat.count[1].counter.value_()
+        if anonrss < 0:
+            anonrss = 0
+        shmrss = task.mm.rss_stat.count[3].counter.value_()
+        if shmrss < 0:
+            shmrss = 0
+        rss_mm = filerss + anonrss
```

Comment:
The indexes you're using... 0, 1, 3. They are named enums in the kernel:
```c
43  enum {
44  	MM_FILEPAGES,	/* Resident file mapping pages */
45  	MM_ANONPAGES,	/* Resident anonymous pages */
46  	MM_SWAPENTS,	/* Anonymous swap entries */
47  	MM_SHMEMPAGES,	/* Resident shared memory pages */
48  	NR_MM_COUNTERS
49  };
```

It would be better to use `prog.constant("MM_FILEPAGES")` instead of "1", just for the principle of it. It helps the reader understand the magic number. Also, by doing `prog.constant("MM_SHMEMPAGES")`, you could detect whether that constant exists (it doesn't on UEK4) which may have been one of the exceptions you were trying to catch with the `except Exception`.

Reply:
Thanks Stephen for pointing out this. I have incorporated these changes in new code.